modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -253,6 +253,76 @@ Test_ProcessPriorityClassAlignment(void)
     free(ProcPriority);
 }
 
+static
+void
+Test_ProcessWx86Information(void)
+{
+    NTSTATUS Status;
+    ULONG VdmPower, ReturnLength;
+
+    /* Everything is NULL */
+    Status = NtQueryInformationProcess(NULL,
+                                       ProcessWx86Information,
+                                       NULL,
+                                       0,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Given an invalid process handle */
+    Status = NtQueryInformationProcess(NULL,
+                                       ProcessWx86Information,
+                                       &VdmPower,
+                                       sizeof(VdmPower),
+                                       NULL);
+    ok_hex(Status, STATUS_INVALID_HANDLE);
+
+    /* Don't query anything */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessWx86Information,
+                                       NULL,
+                                       sizeof(VdmPower),
+                                       NULL);
+    ok_hex(Status, STATUS_ACCESS_VIOLATION);
+
+    /* The buffer is misaligned and information length is wrong */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessWx86Information,
+                                       (PVOID)1,
+                                       0,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* The buffer is misaligned */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessWx86Information,
+                                       (PVOID)1,
+                                       sizeof(VdmPower),
+                                       NULL);
+    ok_hex(Status, STATUS_DATATYPE_MISALIGNMENT);
+
+    /* The buffer is misaligned -- try with an alignment size of 2 */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessWx86Information,
+                                       (PVOID)2,
+                                       sizeof(VdmPower),
+                                       NULL);
+    ok_hex(Status, STATUS_DATATYPE_MISALIGNMENT);
+
+    /* Query the VDM power */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessWx86Information,
+                                       &VdmPower,
+                                       sizeof(VdmPower),
+                                       &ReturnLength);
+    ok_hex(Status, STATUS_SUCCESS);
+    ok(ReturnLength != 0, "ReturnLength shouldn't be 0!\n");
```

Comment:
Does the value vary, or why is the check so generic? I'd assume it'd always be `sizeof(VdmPower)`?

Reply:
My idea is to inspect the internal behaviour of `NtQueryInformationXxx` when returning the length in the last argument call across different information classes. In this case, the length of the requested information is the same like `sizeof(VdmPower)`.

If that test check is too redundant I can simply remove it from the testcase code.